root/static/scripts/common/components/Autocomplete2/types.js

```diff
@@ -86,14 +86,73 @@ export type Actions =
 export type ActionItem = {|
   +action: Actions,
   +id: number | string,
+  +level?: number,
   +name: string | () => string,
   +separator?: boolean,
 |};
 
-export type EntityItem = {|
-  +id: number | string,
-  +level?: number,
-  +name: string,
+type Appearances<T> = {|
+  +hits: number,
+  +results: $ReadOnlyArray<T>,
+|};
+
+export type AutocompleteAreaT = {|
+  ...AreaT,
+  +primaryAlias: string | null,
+|};
```

Comment:
Would it be easier to have `+entity: T` like for `SearchResults<T>` instead of extending `AreaT`?

Reply:
Well, these types just model the format outputted by `/ws/js`, which I'm not really inclined to change on the server at this point. What seems nicer to me is (eventually) just returning normal `AreaT`s etc. from the server and computing `primaryAlias` in JavaScript.